# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.cpp
# Commit: 9896c12c4907
# Full Hash: 9896c12c490709e214030cd99f598e1ffa0076de
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2021-12-17 03:51:53
# Regressor Bug: 1742865
# File Overlap Count: 1
# Description:
#   Bug 1742865 - Handle meta refresh correctly with session history in the parent. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132837
# ==============================================================================

diff -r 607f12999b71 -r 9896c12c4907 docshell/base/nsDocShell.cpp
--- a/docshell/base/nsDocShell.cpp	Thu Dec 16 15:30:29 2021 +0000
+++ b/docshell/base/nsDocShell.cpp	Thu Dec 16 15:31:56 2021 +0000
@@ -8940,8 +8940,11 @@
            this, mLoadingEntry->mInfo.GetURI()->GetSpecOrDefault().get()));
       bool hadActiveEntry = !!mActiveEntry;
       mActiveEntry = MakeUnique<SessionHistoryInfo>(mLoadingEntry->mInfo);
+      // We're passing in mCurrentURI, which could be null. SessionHistoryCommit
+      // does require a non-null uri if this is for a refresh load of the same
+      // URI, but in that case mCurrentURI won't be null here.
       mBrowsingContext->SessionHistoryCommit(
-          *mLoadingEntry, mLoadType, hadActiveEntry, true, true,
+          *mLoadingEntry, mLoadType, mCurrentURI, hadActiveEntry, true, true,
           /* No expiration update on the same document loads*/
           false);
       // FIXME Need to set postdata.
@@ -13472,8 +13475,13 @@
     MOZ_ASSERT(loadingEntry);
     uint32_t loadType =
         mLoadType == LOAD_ERROR_PAGE ? mFailedLoadType : mLoadType;
-    mBrowsingContext->SessionHistoryCommit(
-        *loadingEntry, loadType, hadActiveEntry, aPersist, false, aExpired);
+
+    // We're passing in mCurrentURI, which could be null. SessionHistoryCommit
+    // does require a non-null uri if this is for a refresh load of the same
+    // URI, but in that case mCurrentURI won't be null here.
+    mBrowsingContext->SessionHistoryCommit(*loadingEntry, loadType, mCurrentURI,
+                                           hadActiveEntry, aPersist, false,
+                                           aExpired);
   }
 }
 